cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -390,3 +398,25 @@ func (p *pod) KillTask(ctx context.Context, tid, eid string, signal uint32, all
 	})
 	return eg.Wait()
 }
+
+func (p *pod) DeleteTask(ctx context.Context, tid string) error {
+	t, err := p.GetTask(tid)
+	if err != nil {
+		return errors.Wrap(err, "could not delete task")
+	}
+
+	// although deleting the sandbox task is a no-op, still check that it is not running
+	e, err := t.GetExec("")
```

Comment:
I'm a bit confused on this comment. Empty string denotes that we're asking for the init exec right? If the task we're asking to delete isn't the sandbox task (tid != p.id) then while I agree we should check if any task we're trying to delete isn't running, that behavior shouldn't be exclusive to just the sandbox task.

Reply:
That comment should be moved up earlier, basically this entire operation, `DeleteTask`, is a NoOp if we are deleting the init task. Though, I think based on other conversations, that should change.
We check that all tasks we are trying to delete, sandbox or not, are stopped, so the only way to do that is to look at the init exec (`eid == ""`)